lib/run_part.c

```diff
@@ -30,7 +34,11 @@ int run_part (char *script_path, const char *name, const char *action)
 	if (pid==0) {
 		setenv ("ACTION",action,1);
 		setenv ("SUBJECT",name,1);
-		execv (script_path,args);
+#ifdef HAVE_EXECVEAT
+		execveat (script_fd, "", args, environ, AT_EMPTY_PATH);
+#else
```

Comment:
fexecve(3) is guaranteed by POSIX.1-2008, and is equivalent to `execveat(..., "", ..., AT_EMPTY_PATH)`.  I suggest using fexecve(3).

Reply:
For example OpenBSD does neither support `execveat(2)` nor `fexecve(3)`, is unconditional usage fine?

Reply:
fexecve(3) can be trivially implemented as a wrapper around execve(2).  If some system needs to port shadow somewhere where there's no fexecve(3), we could add our own implementation.

```c
int
fexecve(int fd, char *const argv[], char *const envp[])
{
	char  path[PATH_MAX];

	if (SNPRINTF(path, "/proc/self/fd/%d", fd) == -1)
		return -1;

	return execve(path, argv, envp);
}
```
(I wrote it quickly for this reply; I may have forgotten some details, and it might be buggy.)

Reply:
But I would not add that fallback implementation until someone actually tries to port shadow to such a system.

Reply:
changed